geode-core/src/test/java/org/apache/geode/management/internal/cli/commands/DiskStoreCommandsDUnitTest.java

```diff
@@ -12,1480 +12,283 @@
-      assertEquals(Result.Status.OK, cmdResult.getStatus());
-      String stringResult = commandResultToString(cmdResult);
-      SnapshotTestUtil.checkSnapshotEntries(exportDir, entries, diskStoreName1, region1);
-      SnapshotTestUtil.checkSnapshotEntries(exportDir, entries, diskStoreName1, region2);
-
-    } else {
-      getLogWriter().info("testExportOfflineDiskStore cmdResult is null");
-      fail("Did not get CommandResult in testExportOfflineDiskStore");
-    }
-  }
+    server2.invoke(LocatorServerStartupRule::stopMemberInThisVM);
 
-  /**
-   * Asserts that creating and destroying disk stores correctly updates the shared configuration.
-   */
-  @Test // FlakyTest: GEODE-1406
-  public void testCreateDestroyUpdatesSharedConfig() {
-    disconnectAllFromDS();
-    final int[] ports = AvailablePortHelper.getRandomAvailableTCPPorts(2);
-    jmxPort = ports[0];
-    httpPort = ports[1];
-    try {
-      jmxHost = InetAddress.getLocalHost().getHostName();
-    } catch (UnknownHostException ignore) {
-      jmxHost = "localhost";
-    }
-
-    final String groupName = "testDiskStoreSharedConfigGroup";
-    final String diskStoreName = "testDiskStoreSharedConfigDiskStore";
-
-    // Start the Locator and wait for shared configuration to be available
-    final int locatorPort = AvailablePort.getRandomAvailablePort(AvailablePort.SOCKET);
-    final Properties locatorProps = new Properties();
-    locatorProps.setProperty(NAME, "Locator");
-    locatorProps.setProperty(MCAST_PORT, "0");
-    locatorProps.setProperty(LOG_LEVEL, "fine");
-    locatorProps.setProperty(ENABLE_CLUSTER_CONFIGURATION, "true");
-    locatorProps.setProperty(JMX_MANAGER, "true");
-    locatorProps.setProperty(JMX_MANAGER_START, "true");
-    locatorProps.setProperty(JMX_MANAGER_BIND_ADDRESS, String.valueOf(jmxHost));
-    locatorProps.setProperty(JMX_MANAGER_PORT, String.valueOf(jmxPort));
-    locatorProps.setProperty(HTTP_SERVICE_PORT, String.valueOf(httpPort));
-
-    Host.getHost(0).getVM(0).invoke(new SerializableRunnable() {
-      @Override
-      public void run() {
-        final File locatorLogFile = new File("locator-" + locatorPort + ".log");
-        try {
-          final InternalLocator locator = (InternalLocator) Locator.startLocatorAndDS(locatorPort,
-              locatorLogFile, null, locatorProps);
-
-          WaitCriterion wc = new WaitCriterion() {
-            @Override
-            public boolean done() {
-              return locator.isSharedConfigurationRunning();
-            }
-
-            @Override
-            public String description() {
-              return "Waiting for shared configuration to be started";
-            }
-          };
-          waitForCriterion(wc, 5000, 500, true);
-        } catch (IOException ioex) {
-          fail("Unable to create a locator with a shared configuration");
-        }
-      }
-    });
+    props.setProperty("log-file", "");
+    rule.startServerVMAsync(1, props, locator.getPort());
```

Comment:
Although more verbose, I think that rewriting this as
{{rule.startServerVMAsync(server1.getVM().getId(), props, locator.getPort());}}
makes the intent clearer.

Or better yet, add a `startServerVMAsync` method that accepts `MemberVM` instead of an int.

Reply:
I think your former suggestion is good. In this case it's a restart but at other times we may not have an existing `MemberVM` available.

Reply:
To clarify my 2nd suggestion, I would keep the index based methods and add the MemberVM based method for restarts. Or not, there will probably never be that many restart situations so just following my 1st suggestion should be enough;